spec/event/batch_event_processor_spec.rb

```diff
@@ -0,0 +1,244 @@
+# frozen_string_literal: true
+
+#
+#    Copyright 2019, Optimizely and contributors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+require 'spec_helper'
+require 'optimizely/event/batch_event_processor'
+require 'optimizely/event/user_event_factory'
+require 'optimizely/exceptions'
+require 'optimizely/event_dispatcher'
+require 'optimizely/error_handler'
+require 'optimizely/helpers/constants'
+require 'optimizely/helpers/validator'
+require 'optimizely/logger'
+describe Optimizely::BatchEventProcessor do
+  WebMock.allow_net_connect!
+  let(:config_body_JSON) { OptimizelySpec::VALID_CONFIG_BODY_JSON }
+  let(:error_handler) { Optimizely::NoOpErrorHandler.new }
+  let(:spy_logger) { spy('logger') }
+  let(:project_config) { Optimizely::DatafileProjectConfig.new(config_body_JSON, spy_logger, error_handler) }
+  let(:event) { project_config.get_event_from_key('test_event') }
+  let(:log_url) { 'https://logx.optimizely.com/v1/events' }
+  let(:post_headers) { {'Content-Type' => 'application/json'} }
+
+  MAX_BATCH_SIZE = 10
+  MAX_DURATION_MS = 1000
+  TIMEOUT_INTERVAL_MS = 5000
+
+  before(:example) do
+    @event_queue = []
+    @event_dispatcher = Optimizely::EventDispatcher.new
+    allow(@event_dispatcher).to receive(:dispatch_event).with(instance_of(Optimizely::Event))
+  end
+
+  it 'return empty event queue event is processed' do
```

Comment:
Can we also make sure if the dispatchEvent is getting called and logevent is correct?